### PR TITLE
Introduce intradoc-links in a lot of places

### PR DESCRIPTION
**Pull request by kangalio**
I went through the serenity source code using VSCode [regex](https://pastebin.com/raw/28HZKR15) find-and-replace and tried to find and convert as many item references into intradoc links as possible.

There were a few things I am uncertain about:
- should items from std be turned into intradoc-links as well (None, Future, HashSet, File, Path, Result, Vec, String, Arc)?
- should references to `Self` methods be written as
  ```
  Lorem ipsum you can call [`method_name`] to do foo bar.
  ...
  [`method_name`]: Self::method_name
  ```
  or just
  ```
  Lorem ipsum you can call [`Self::method_name`] to do foo bar.
  ```

Regexes used (can't enclose in backquotes because the regexes contain backquotes themselves):
- Initial regex to find all items within code-blocks that are not intradoc-links: (///.*?)([^\\[])(\`[\w:_]+\`)([^\\]]|^) => $1$2[$3]$4
- To find remaining std items, this regex matches camel-case names inside code blocks that are not intradoc links: (///.\*[^\\[])\`([A-Z][a-z][A-Za-z]\*)\`([^\\]]|$) => $1[\`$2\`]$3

### CODE REVIEW

**Comment by arqunis** [comment]
> There were a few things I am uncertain about: 
>
> - should items from std be turned into intradoc-links as well (None, Future, HashSet, File, Path, Result, Vec, String, Arc)?
> - should references to Self methods be written as 

For the first point, yes. 
For the second point, it should be the latter. As far I am aware, links to methods of `Self` are on their own lines in a few places already. Thus, for consistency, the same should be done.

**Comment by Lakelezz** [comment]
I agree with acdenisSK!

File: src/error.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ use crate::model::ModelError;
 pub type Result<T> = StdResult<T, Error>;
 
 /// A common error enum returned by most of the library's functionality within a
-/// custom [`Result`].
+/// custom [[`Result`]].
```
Comment (nickelc) [inline src/error.rs:39]: this looks wrong

Comment (nickelc) [inline src/error.rs]:
```suggestion
/// custom [`Result`].
```

Comment (kangalio) [reply src/error.rs]: Thanks for finding those two cases

File: src/model/channel/channel_id.rs [hunk 2]
```diff
@@ -710,6 +710,7 @@ impl ChannelId {
     /// [`CreateMessage::content`]: crate::builder::CreateMessage::content
     /// [Attach Files]: Permissions::ATTACH_FILES
     /// [Send Messages]: Permissions::SEND_MESSAGES
+    /// [`File`]: std::fs::File
```
Comment (nickelc) [inline src/model/channel/channel_id.rs]: Tokio's `File` is used for `Into<AttachmentType>`.  See example.
```suggestion
    /// [`File`]: tokio::fs::File
```

**Review by arqunis — APPROVED** [review]

**Pull request merged**